### PR TITLE
Cygwin fix for SM4 (issue #663)

### PR DESCRIPTION
**Pull request by karel-m**
Fix for #663
* Static functions renamed (using `s_` prefix) so that they do not collide with cygwin stuff
* Added missing `#ifdef LTC_TEST`

### CODE REVIEW

File: src/ciphers/sm4.c [hunk 1]
```diff
@@ -67,7 +67,7 @@ static const sm4_u8_t sbox_table[16][16] = {
  * S-box
  * defined in section 2.6 S-box
  */
-LTC_INLINE static sm4_u8_t sbox(sm4_u8_t a)
+LTC_INLINE static sm4_u8_t s_sbox(sm4_u8_t a)
```
Comment (sjaeckel) [inline src/ciphers/sm4.c]: I'd suggest to simply namespace all the functions, but leave their original name intact.

```suggestion
LTC_INLINE static sm4_u8_t s_sm4_sbox(sm4_u8_t a)
```

File: src/ciphers/sm4.c [hunk 2]
```diff
@@ -80,35 +80,35 @@ LTC_INLINE static sm4_u8_t sbox(sm4_u8_t a)
  * But we just convert a 32bit word byte by byte.
  * So it's OK if we don't convert the endian order
  */
-LTC_INLINE static sm4_u32_t t(sm4_u32_t A)
+LTC_INLINE static sm4_u32_t s_trans(sm4_u32_t A)
```
Comment (sjaeckel) [inline src/ciphers/sm4.c]:
```suggestion
LTC_INLINE static sm4_u32_t s_sm4_t(sm4_u32_t A)
```

and so on.

**Review by sjaeckel — CHANGES_REQUESTED** [review]
Like that we also would prevent name clashes as seen when trying to do #596 and the names are still kind of the same from the original and not something entirely different.

File: src/ciphers/sm4.c (hunk 2 above)
Comment (karel-m) [reply src/ciphers/sm4.c]: done

File: src/ciphers/sm4.c (hunk 1 above)
Comment (karel-m) [reply src/ciphers/sm4.c]: done

**Review by levitte — APPROVED** [review]
LGTM

**Review by sjaeckel — APPROVED** [review]

**Pull request merged**